### PR TITLE
[MSHARED-871] add basic documentation

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

**Review by slachiewicz — APPROVED** [review]

File: src/site/markdown/index.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# About Apache Verifier Maven Component
+
+Provides a test harness for Maven integration tests. This is a library which can be used in own Java-based integration tests. Look at the [Getting Started guide](./getting-started.html) for more information on how to use it.
+
+Other example usages can be found in the integration tests of the [Apache Maven Remote Resources Plugin](https://github.com/apache/maven-remote-resources-plugin/blob/maven-remote-resources-plugin-1.7.0/src/test/java/org/apache/maven/plugin/resources/remote/it/IT_FilterLocalOverride.java).
```
Comment (slawekjaranowski) [inline src/site/markdown/index.md:24]: It is simple test ...
If we want to show more complicated usage I will link to Maven Integration project as reference
https://github.com/apache/maven-integration-testing

File: src/site/markdown/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+# Getting Started
+
+## Overview
+
+Using the `Verifier` consists out of three different phases
+
+1. Configure
+2. Run
+3. Verify
```
Comment (slawekjaranowski) [inline src/site/markdown/getting-started.md:28]: no needed  table of content for short document

File: src/site/markdown/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## Configure
+
+The `Verifier` instance is constructed in the simplest case with just one argument taking the path name of the base directory containing the `pom.xml` of the project you want to build.
+
+```
+String baseDir = ...
+Verifier verifier = new Verifier(baseDir);
```
Comment (slawekjaranowski) [inline src/site/markdown/getting-started.md:36]: please follow maven convention, space after and before parentheses

File: src/site/markdown/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+The configuration can be further tweaked with additional setter methods and/or constructors taking more parameters.
+
+### System Properties
+
+The following systen properties/environment variables are evaluated
+
+| System Property | Description |
```
Comment (slawekjaranowski) [inline src/site/markdown/getting-started.md:45]: Please add to table:
Which properties are required?
What is default values / behavior if property is not set?

File: src/site/markdown/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+Whether the embedded or the forked launcher are used depends on the field `forkJvm` set through the constructor or `setForkJvm` or as fallback on the value of system property `verifier.forkMode`.
+
+#### Determining Maven Home directory
+
+The following directories are considered as potential Maven home directory (relevant for both forked launcher and embedded launcher with  [Plexus Classworlds Loader][plexus-classwords]). The first existing directory terminates the mechanism. 
+
+1. Maven Home path given in the constructor
+2. System property `maven.home`
+3. Environment variable `M2_HOME`
```
Comment (slawekjaranowski) [inline src/site/markdown/getting-started.md:73]: M2_HOME is deprecated - should be removed from code

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]
- do we need some of special dependency for project in order to use embedded mode?
- example of setting mave.home propery by failsafe will be useful, like
```xml
<plugin>
    <artifactId>maven-failsafe-plugin</artifactId>
    <configuration>
        <systemPropertyVariables>
            <maven.home>${maven.home}</maven.home>
        </systemPropertyVariables>
    </configuration>
</plugin>
```

File: src/site/markdown/getting-started.md (hunk 2 above)
Comment (kwin) [reply src/site/markdown/getting-started.md:28]: Well, that is no toc but rather some explanation of the phases in paragraph "Overview".

File: src/site/markdown/getting-started.md (hunk 5 above)
Comment (kwin) [reply src/site/markdown/getting-started.md:73]: Please create a dedicated JIRA issue for that. This PR is just about documenting the status quo.

**Comment by slawekjaranowski** [comment]
pleas also add menu item under `OVERVIEW` section.

**Review by slawekjaranowski — APPROVED** [review]

File: src/site/markdown/index.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# About Apache Verifier Maven Component
```
Comment (hboutemy) [inline src/site/markdown/index.md:20]: Apache Maven Verifier :)

File: src/site/markdown/index.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+Provides a test harness for Maven integration tests. This is a library which can be used in own Java-based integration tests. Look at the [Getting Started guide](./getting-started.html) for more information on how to use it.
+
+More complex example usages can be found in the integration tests of [Apache Maven](https://github.com/apache/maven-integration-testing).
```
Comment (hboutemy) [inline src/site/markdown/index.md:24]: the integration tests of [Apache Maven](https://github.com/apache/maven-integration-testing
=>
the [different integration tests](https://github.com/apache/maven-integration-testing/tree/master/core-it-suite/src/test/java/org/apache/maven/it) of [Maven Core Integration Tests](https://github.com/apache/maven-integration-testing)

File: src/site/markdown/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+```
+String baseDir = ...
+Verifier verifier = new Verifier( baseDir );
+```
+
+The configuration can be further tweaked with additional setter methods and/or constructors taking more parameters.
+
+### System Properties
+
+The following systen properties/environment variables are evaluated. None of them are required and in most cases the same behavior can also be achieved through constructor or setter method programmatically.
```
Comment (hboutemy) [inline src/site/markdown/getting-started.md:43]: s/systen/system/

**Review by hboutemy — APPROVED** [review]
nice work done, thank you
can be merged now or with the few little enhancements proposed

**Comment by jira-importer** [comment]
Resolve #145

**Pull request merged**